tutorials/ui/bbcode_in_richtextlabel.rst

```diff
@@ -9,54 +9,183 @@ Introduction
+property using available tags. Both properties are located at the top of the inspector
+after selecting a RichTextLabel node.
 
-.. image:: img/bbcodeText.png
+.. image:: img/bbcode_in_richtextlabel_inspector.webp
 
-For example, ``BBCode [color=blue]blue[/color]`` would render the word "blue" with
-a blue color.
+For example, ``BBCode [color=green]test[/color]`` would render the word "test" with
+a green color.
 
-.. image:: img/bbcodeDemo.png
+.. image:: img/bbcode_in_richtextlabel_basic_example.webp
 
-Most BBCodes consist of 3 parts: the opening tag, the content and the closing tag. The
-opening tag delimits the start of the formatted part, and can also carry some
-configuration options. Some opening tags, like the ``color`` one shown above, also require
-a value to work. The closing tag delimits the end of the formatted part. In some cases
-both the closing tag and the content can be omitted.
+Most BBCodes consist of 3 parts: the opening tag, the content and the closing
+tag. The opening tag delimits the start of the formatted part, and can also
+carry some configuration options. Some opening tags, like the ``color`` one
+shown above, also require a value to work. Other opening tags may accept
+multiple options (separated by spaces within the opening tag). The closing tag
+delimits the end of the formatted part. In some cases, both the closing tag and
+the content can be omitted.
 
 .. code-block:: none
 
   [tag]content[/tag]
   [tag=value]content[/tag]
-  [tag options]content[/tag]
+  [tag option1=value1 option2=value2]content[/tag]
   [tag][/tag]
   [tag]
 
 .. note::
 
-    In the Inspector you may notice that after writing in the "BBCode Text" property the
-    regular "Text" property now has the same text but without BBCodes. Take care not to
-    edit the regular "Text" property! You will lose the BBCode markup if you do. All changes
-    to the text must be done using the "BBCode Text".
+    RichTextLabel doesn't support entangled BBCode tags. For example, instead of
+    using:
+
+    ::
+
+        [b]bold[i]bold italic[/b]italic[/i]
+
+    Use:
+
+    ::
+
+        [b]bold[i]bold italic[/i][/b][i]italic[/i]
+
+.. _doc_bbcode_in_richtextlabel_handling_user_input_safely:
+
+Handling user input safely
+--------------------------
+
+In a scenario where users may freely input text (such as chat in a multiplayer
+game), you should make sure users cannot use arbitrary BBCode tags that will be
+parsed by RichTextLabel. This is to avoid inappropriate use of formatting, which
+can be problematic if ``[url]`` tags are handled by your RichTextLabel (as players
+may be able to create clickable links to phishing sites or similar).
+
+Using RichTextLabel's ``[lb]`` and/or ``[rb]`` tags, we can replace the opening and/or
+closing brackets of any BBCode tag in a message with those escaped tags. This
+prevents users from using BBCode that will be parsed as tags – instead, the
+BBCode will be displayed as text.
```

Comment:
Perhaps it's worth mentioning the alternative to _remove_ tags with a regular expression?

```gdscript
var regex = RegEx.new()
regex.compile("\\[.*?\\]")
text = regex.sub(text, "", true)
```

Reply:
I'm not sure. In some cases, stripping tags is desired, but for something like multiplayer chat, it can have unintended effects. For example, if I write a message `[IMPORTANT] Hello world`, I don't expect the `[IMPORTANT]` to disappear entirely.